dhall-docs/README.md

```diff
@@ -97,6 +60,407 @@ dhall-docs --input . --output-link ${OTHER_LINK}
+you separate your internal block comments from the ones that you want
+`dhall-docs` to output in the final markup.
+
+This is an example of a file documented using block comments:
+
+```dhall
+{-|
+# My awesome package
+
+This is the header of my package
+
+* a list item in **bold**
+  - a sub item in _italic_
+  - a [link](https://example.com)
+    * a `code` quote
+
+
+A new paragraph
+
+    an indented code block
+-}
+let myAwesomeFunction {- this will be ignored on the documentation -}
+               = \(x : Natural) -> x + 1
+
+in
+{
+  {-|
+  foo
+      bar
+          baz
+  -}
+  myAwesomeFunction
+}
+```
+
+### Line comments
+
+Normal Dhall single-line comments start with `--` and ends at a newline. For
+example:
+
+```dhall
+-- single-line comment
+0
+```
+
+A `dhall-docs` valid single-line comment starts with `--|` and a single whitespace
+character. For example:
+
+```dhall
+--| foo
+```
+
+You can span your documentation in several single-line comments by writing `--` and two
+(2) whitespaces:
+
+```dhall
+--| foo
+--  bar
+--      baz
+```
+
+The `--` prefix defines the base indentation for any line. That is, on the above
+example `dhall-docs` will extract the following contents:
+
+```
+foo
+bar
+    baz
+```
+
+If you would like to force an empty line on the documentation output, place an
+empty single line comment in that place. On this example:
+
+```dhall
+--| foo
+--
+--  bar
+```
+
+`dhall-docs` will extract:
+
+```dhall
+foo
+
+bar
+```
+
+The empty `--` line doesn't need the two (2) whitespaces.
+
+If between two single-line comments there is one (1) or more empty lines, the
+latter comments will be ignored. On this example:
+
+
+```dhall
+--| foo 
+--  bar
+
+--  baz
+```
+
+`dhall-docs` will only capture this text:
+
+```
+foo
+bar
+```
+
+In a set of subsequent single-line comments, all lines before the one that has
+the `|` character will be ignored, meaning that if any line on the set doesn't
+have the `|` marker, no text will be extracted for the generated documentation.
+
+On this example:
+
+```dhall
+--  foo
+--  bar
+--| baz
+--  qux
+```
+
+the following text will be extracted:
+
+```dhall
+baz
+qux
+```
+
+and this set of single-line comments:
+
+```dhall
+--  foo
+--  bar
+--  baz
+```
+
+will be ignored by the tool.
+
+The set of `--` lines needs to be aligned with the same number of preceding
+characters, meaning that a set of lines like this:
+
+```dhall
+--| foo
+  --  bar
+ --  baz
+```
+
+is invalid: `dhall-docs` will ignore this on the generated documentation and
+a warning will be logged on the console.
+
+To end, if there are language tokens in-between a set of lines, the lines after
+those tokens will be ignored. On this example:
+
+```dhall
+--| foo
+,
+--  bar
+```
+
+the extracted text will be the following:
+
+    foo
+
+Here is an example of using this type of comments in a Dhall file:
+
+```dhall
+--| # My header
+--
+--  * 1
+--  * 2
+
+let a --| foo
+      --  bar
+        = True
```

Comment:
```suggestion
let --| foo
    --  bar
    a = True
```

I mainly suggest this since that's the commenting format we've been recommending so far for `let` bindings